modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -69,18 +90,57 @@ class LsfExecutor extends AbstractGridExecutor {
             // is divided by the number of used cpus (processes)
             if( task.config.cpus > 1 && !perJobMemLimit ) {
                 long bytes = mem.toBytes().intdiv(task.config.cpus as int)
-                result << '-M' << String.valueOf(MemoryUnit.of(bytes).toMega())
+                // patch in order to check memory units
+                // check memory_lsf_unit_for_limits whether ['KB', 'MB', 'GB', 'TB']
+                if (memory_lsf_unit_for_limits.toUpperCase() in ['KB']){
+                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toKilo())
+                    result << '-R' << "select[mem>=${mem.toKilo()}] rusage[mem=${mem.toKilo()}]".toString()
+                }
+                else if (memory_lsf_unit_for_limits.toUpperCase() in ['MB']){
+                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toMega())
+                    result << '-R' << "select[mem>=${mem.toMega()}] rusage[mem=${mem.toMega()}]".toString()
+                }
+                else if (memory_lsf_unit_for_limits.toUpperCase() in ['GB']){
+                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toGiga())
+                    result << '-R' << "select[mem>=${mem.toGiga()}] rusage[mem=${mem.toGiga()}]".toString()
+                }
+                else { 
+                    new throw Exception ("LSF_UNIT_FOR_LIMITS not in KB, MB, or GB. Check variable $LSF_ENVDIR/lsf.conf")
+                }
```

Comment:
Ummmm, too many `if`, I would suggests to isolate this into a separate function taking the the MemUnit value of the unit string and returning the expected value. It could even be a nice method to be added to the [MemoryUnit](https://github.com/nextflow-io/nextflow/blob/73977a96daedcbe59a3b1d7cbeb21b67156bc539/modules/nf-commons/src/main/nextflow/util/MemoryUnit.groovy#L32) class it self.

Reply:
I understand your point now, see below

Reply:
Still too many :) 

The logic should be identical, having a`.toUnit()` in the MemoryUnity unit class, it's enough to replace `mem.toMega()` with `mem.toUnit(memoryUnitLSF)`